hack/common.sh

```diff
@@ -217,6 +229,16 @@ os::build::build_binaries() {
           "$(dirname ${test})"
       done
 
+      for artifact in ${binaries[@]}; do
```

Comment:
check if the split between nonstatics and tests above already segments out your e2e binary... this loop may not even be needed any more